src/settings.rs

```diff
@@ -106,11 +108,30 @@ impl Default for Settings {
             test_mode: false,
             test_file_path: "./tools/test/test_data/".to_owned(),
             location_test_header: None,
+            default_location: "USOK".to_owned(),
         }
     }
 }
 
 impl Settings {
+    pub fn verify_settings(&mut self) -> Result<(), ConfigError> {
+        if self.adm_endpoint_url.is_empty() {
+            return Err(ConfigError::Message("Missing adm_endpoint_url".to_owned()));
+        }
+        self.default_location = self.default_location.to_uppercase();
+        self.default_location.retain(|e| e.is_alphabetic());
+        if self.default_location.len() < 4 {
+            return Err(ConfigError::Message(
+                "Invalid default location specified. Please use a string like \"USOK\"".to_owned(),
+            ));
+        };
+        self.default_location = self.default_location[..4].to_owned();
```

Comment:
if we're clipping the end to 4 I think we should call anything larger invalid.

Reply:
Apparently 3166-2 (subdivision) is potentially 3 alphanumeric characters: 

https://en.wikipedia.org/wiki/ISO_3166-2

I was tempted to suggest the comma format but that doesn't matter. 

3166-1 alpha-2 (country) is two letter: https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2